src/nms-api/IMessageProducer.cs

```diff
@@ -117,41 +117,81 @@ public interface IMessageProducer : System.IDisposable
         /// </summary>
         IMessage CreateMessage();
 
+        /// <summary>
+        /// Creates a new message with an empty body
+        /// </summary>
+        Task<IMessage> CreateMessageAsync();
```

Comment:
What's the expectation for non async implementations? I mean, what if async implementation doesn't make sense? Are we going to throw NotSupportedException or should we return the result of the sync equivalent? If the latter, I'd encourage you to use ValueTasks instead.